tensorboard/webapp/runs/store/runs_reducers.ts

```diff
@@ -98,26 +99,38 @@ const dataReducer: ActionReducer<RunsDataState, Action> = createReducer(
       };
     }
 
-    const eidsBasedKey = serializeExperimentIds(action.experimentIds);
-    if (!nextSelectionState.has(eidsBasedKey)) {
-      const selectionMap = new Map<string, boolean>();
-      const runSelected =
-        action.runsForAllExperiments.length <=
-        MAX_NUM_RUNS_TO_ENABLE_BY_DEFAULT;
-      for (const run of action.runsForAllExperiments) {
-        selectionMap.set(run.id, runSelected);
+    /**
+     * Auto-select new runs based on the saved preference, or based on the
+     * number of runs.
+     *
+     * Known caveat: subsequent reloads after the first may result
+     * in selecting a large number of runs automatically, incuring a
+     * performance cost.
+     */
+    let nextShouldAutoSelectRunsMap = state.shouldAutoSelectRuns;
+    const runCount = action.runsForAllExperiments.length;
+    if (runCount > 0) {
+      const eidsBasedKey = serializeExperimentIds(action.experimentIds);
+      let selectNewRunsByDefault =
```

Comment:
Didn't think that this variable would be a boolean type. Perhaps `shouldSelectNewRuns`?